src/WebJobs.Extensions.DurableTask/HttpApiHandler.cs

```diff
@@ -203,6 +220,17 @@ public HttpApiHandler(DurableTaskExtension config, TraceWriter traceWriter)
             return response;
         }
 
+        private static bool GetQueryParameterValue(HttpRequestMessage request, string queryParameterName)
+        {
+            if (!request.GetQueryNameValuePairs().AllKeys.Contains(queryParameterName))
+            {
+                return false;
+            }
+
+            var value = request.GetQueryNameValuePairs()[queryParameterName];
```

Comment:
`GetQueryNameValuePairs` parses the query string every time it is called, right? The `GetQueryParameterValue` method is called several times in a row. Can we refactor this so that we only call it once and cache the result so we can quickly look up all the different query string values we care about without reparsing?

Reply:
@cgillum Sure. I like very much this optimization! I will work on it!